internal/controller/isbservicerollout_controller.go

```diff
@@ -336,30 +345,15 @@ func (r *ISBServiceRolloutReconciler) getRolloutKey(rolloutNamespace string, rol
 }
 
 // return:
-// - whether ISBService needs to update
 // - whether it's in the process of being updated
 // - error if any
-func (r *ISBServiceRolloutReconciler) isISBServiceUpdating(ctx context.Context, isbServiceRollout *apiv1.ISBServiceRollout, existingISBSVCDef *kubernetes.GenericObject) (bool, bool, error) {
-
+func (r *ISBServiceRolloutReconciler) isISBServiceUpdating(ctx context.Context, existingISBSVCDef *kubernetes.GenericObject) (bool, error) {
 	isbServiceReconciled, _, err := r.isISBServiceReconciled(ctx, existingISBSVCDef)
 	if err != nil {
-		return false, false, err
-	}
-
-	existingSpecAsMap := make(map[string]interface{})
-	err = json.Unmarshal(existingISBSVCDef.Spec.Raw, &existingSpecAsMap)
-	if err != nil {
-		return false, false, err
-	}
-	newSpecAsMap := make(map[string]interface{})
-	err = util.StructToStruct(&isbServiceRollout.Spec.InterStepBufferService.Spec, &newSpecAsMap)
-	if err != nil {
-		return false, false, err
+		return false, err
 	}
-	// TODO: see if we can just use DeepEqual or if there may be null values we need to ignore
-	isbServiceNeedsToUpdate := !util.CompareMapsIgnoringNulls(existingSpecAsMap, newSpecAsMap)
 
-	return isbServiceNeedsToUpdate, !isbServiceReconciled, nil
+	return !isbServiceReconciled, nil
```

Comment:
this is good having this function truly just being `isISBServiceUpdating()` - it wasn't great that it represented both things before